### ISSUE TITLE
_MaterialState#77020(tickers: tracking 1 ticker) was disposed with an active Ticker. (I didnt use a controller )

### ISSUE THREAD

**Issue by jpainam** [flutter/flutter#49814]
Using a InkWell causes this error. I don't have any controller in the page.

```dart
════════ Exception caught by widgets library ═══════════════════════════════════════════════════════
The following assertion was thrown while finalizing the widget tree:
_MaterialState#77020(tickers: tracking 1 ticker) was disposed with an active Ticker.

_MaterialState created a Ticker via its TickerProviderStateMixin, but at the time dispose() was called on the mixin, that Ticker was still active. All Tickers must be disposed before calling super.dispose().

Tickers used by AnimationControllers should be disposed by calling dispose() on the AnimationController itself. Otherwise, the ticker will leak.

The offending ticker was: _WidgetTicker(created by _MaterialState#77020(tickers: tracking 2 tickers))
The stack trace when the _WidgetTicker was actually created was:
#0      new Ticker.<anonymous closure> (package:flutter/src/scheduler/ticker.dart:66:40)
#1      new Ticker (package:flutter/src/scheduler/ticker.dart:68:6)
#2      new _WidgetTicker (package:flutter/src/widgets/ticker_provider.dart:237:80)
#3      TickerProviderStateMixin.createTicker (package:flutter/src/widgets/ticker_provider.dart:168:34)
#4      new AnimationController (package:flutter/src/animation/animation_controller.dart:245:21)
#5      new MaterialInkSplash (package:maraka/widgets/material_ink_splash.dart:161:30)
#6      MaterialInkSplashFactory.create (package:maraka/widgets/material_ink_splash.dart:66:12)
#7      _InkResponseState._createInkFeature (package:flutter/src/material/ink_well.dart:625:72)
#8      _InkResponseState._startSplash (package:flutter/src/material/ink_well.dart:691:42)
#9      _InkResponseState._handleTapDown (package:flutter/src/material/ink_well.dart:674:5)
#10     TapGestureRecognizer.handleTapDown.<anonymous closure> (package:flutter/src/gestures/tap.dart:463:51)
#11     GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:182:24)
#12     TapGestureRecognizer.handleTapDown (package:flutter/src/gestures/tap.dart:463:11)
#13     BaseTapGestureRecognizer._checkDown (package:flutter/src/gestures/tap.dart:256:5)
#14     BaseTapGestureRecognizer.acceptGesture (package:flutter/src/gestures/tap.dart:234:7)
#15     GestureArenaManager.sweep (package:flutter/src/gestures/arena.dart:156:27)
#16     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:222:20)
#17     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:198:22)
#18     GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:156:7)
#19     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:102:7)
#20     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:86:7)
#24     _invoke1 (dart:ui/hooks.dart:273:10)
#25     _dispatchPointerDataPacket (dart:ui/hooks.dart:182:5)
(elided 3 frames from package dart:async)

When the exception was thrown, this was the stack: 
#0      TickerProviderStateMixin.dispose.<anonymous closure> (package:flutter/src/widgets/ticker_provider.dart:185:13)
#1      TickerProviderStateMixin.dispose (package:flutter/src/widgets/ticker_provider.dart:203:6)
#2      StatefulElement.unmount (package:flutter/src/widgets/framework.dart:4435:12)
#3      _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1748:13)
#4      ListIterable.forEach (dart:_internal/iterable.dart:41:13)
...
════════════════════════════════════════════════════════════════════════════════════════════════════

════════ (2) Exception caught by widgets library ═══════════════════════════════════════════════════
_MaterialState#df654(tickers: tracking 1 ticker) was disposed with an active Ticker.
════════════════════════════════════════════════════════════════════════════════════════════════════
```
## Steps to Reproduce

1. Run `flutter create bug`.
2. Update the files as follows: ... 
3. ... 

**Expected results:** 

**Actual results:** 

<details>
  <summary>Logs</summary>

```
```

```
```

```
[√] Flutter (Channel stable, v1.12.13+hotfix.5, on Microsoft Windows [Version 10.0.18362.592], locale en-GB)
    • Flutter version 1.12.13+hotfix.5 at D:\flutter\flutter
    • Framework revision 27321ebbad (7 weeks ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0
```

</details>

**Comment by iapicca** [comment]
Hi @jpainam 
can you please provide a reproducible minimal code sample.
Thank you

**Comment by jpainam** [comment]
This is actually a mininal example of what i'm achieved. i removed `fetching the data online` and everything seems to be working fine. I will check my code again. thank you
```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {



  Map<String, List<String>> data = const {
    "key1": ["data 1.1", "data 1.2", "data 1.3"],
    "key2": ["data 2.1", "data 2.2", "data 2.3"],
    "key3": ["data 3.1", "data 3.2", "data 3.3"]
  };
  List<String> subCategories = [];
  @override
  void initState() {
    super.initState();
    subCategories.addAll(data[data.keys.first]);
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Row(
        children: <Widget>[
          Expanded(
            flex: 2,
            child: ListView.builder(
              itemBuilder: (_, index) => _buildItem(_, index, data.keys.elementAt(index)),
              itemCount: data.keys.length,
              physics: const AlwaysScrollableScrollPhysics(),
            ),
          ),
          subCategories.length !=0 ? Expanded(
            flex: 5,
            child: GridView.builder(
              itemCount: subCategories.length,
              gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
                  crossAxisCount: 2
              ),
              itemBuilder: (context, index){
                return InkWell(onTap: () {

                }, child: Text(subCategories.elementAt(index)));
              },
            ),
          ) : SizedBox(),
        ],
      ),
    );
  }
  

  static const double kDropdownMenuItemHeight = 50.0;

  int _activeIndex = 0;

  Widget _buildItem(BuildContext context, int index, String key) {
    return index != _activeIndex
        ? DecoratedBox(
            decoration: new BoxDecoration(
              color: Colors.grey[100],
              border: new Border(
                bottom: Divider.createBorderSide(context),
                top: Divider.createBorderSide(context),
              ),
            ),
            child: Material(
              color: Colors.transparent,
              child: InkWell(
                onTap: () {
                  setState(() {
                
                    subCategories.clear();
                    subCategories.addAll(data[key]);
                      _activeIndex = index;
                  });
                },
                child: Padding(
                  padding: const EdgeInsets.only(left: 8.0),
                  child: Row(
                    mainAxisAlignment: MainAxisAlignment.center,
                    crossAxisAlignment: CrossAxisAlignment.center,
                    children: <Widget>[
                      Expanded(
                        child: Text(
                          key,
                        ),
                      ),
                    ],
                  ),
                ),
              ),
            ),
          )
        : DecoratedBox(
            decoration: BoxDecoration(
                color: Colors.white,
                border: Border(
                    top: Divider.createBorderSide(context),
                    bottom: Divider.createBorderSide(context))),
            child: Container(
              child: Row(
                mainAxisAlignment: MainAxisAlignment.center,
                crossAxisAlignment: CrossAxisAlignment.center,
                children: <Widget>[
                  Container(
                      color: Colors.pink[400],
                      width: 3.0,
                      height: kDropdownMenuItemHeight),
                  Expanded(
                    child: Padding(
                      padding: const EdgeInsets.only(left: 5.0),
                      child: Text(
                        key,
                      ),
                    ),
                  ),
                ],
              ),
            ));
  }
}
```

**Comment by jpainam** [comment]
In `material_ink_splash.dart`, `_alphaFadeInController` is never dispose in ` _handleAlphaFadeInStatusChanged` if the `fading animation` is never  completed. I noticed that. when i hold down the `InkWell onTap` and complete the fading animation, this error `_MaterialState#2ac06(tickers: tracking 1 ticker) was disposed with an active Ticker.` is not shown. 

I got the error. Was using a custom `splashFactory: InkSplash.splashFactory` to speed up the fading effect. The current fading effect on InkWell doesn't work quite well as for the native app.
